src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js

```diff
@@ -0,0 +1,128 @@
+import classNames from 'classnames';
+import getProp from 'lodash/get';
+import noop from 'lodash/noop';
+
+import InlineDelete from '../inline-delete';
+import { ReadableTime } from '../../../../components/time';
+import Tooltip from '../../../../components/tooltip';
+import { Link } from '../../../../components/link';
+import messages from '../../../common/messages';
+import { ONE_HOUR_MS } from '../../../../constants';
+
+import './AppActivity.scss';
+
+const parser = new DOMParser();
+
+type Props = {
+    activity_template: Object,
+    app: Object,
+    created_at: string,
+    created_by: User,
+    currentUser: User,
+    error?: ActionItemError,
+    id: string,
+    isPending: boolean,
+    onDelete: Function,
+    permissions?: BoxItemPermission,
+    rendered_text: string,
+};
+
+function parseActivity(htmlString: string): Array<HTMLLinkElement> {
+    const doc = parser.parseFromString(htmlString, 'text/html');
+    // $FlowFixMe
+    return Array.from(doc.body.childNodes);
+}
+
+function mapActivityNodes(node: HTMLLinkElement): React.Node {
+    const { dataset = {}, href = '#', tagName, textContent } = node;
+    switch (tagName) {
+        case 'A':
+            return (
+                <Link
+                    href={href}
+                    data-resin-target={dataset.resinTarget}
+                    data-resin-action={dataset.resinAction}
+                    key={`app_actvity_link_${href}`}
+                    rel="roreferrer noopener"
+                    className="bcs-app-activity-link"
+                >
+                    {textContent}
+                </Link>
+            );
+        default:
+            return textContent;
+    }
+}
+
+const AppActivity = ({
+    activity_template,
+    app,
+    currentUser,
+    created_at,
+    created_by,
+    error,
+    id,
+    isPending,
+    rendered_text,
+    permissions = {},
+    onDelete = noop,
+}: Props): React.Node => {
+    const createdAtTimestamp = new Date(created_at).getTime();
+    const activityNodes = parseActivity(rendered_text);
+    const canDelete = getProp(permissions, 'can_delete', false);
+    const { name, icon_url } = app;
+    const { id: templateId } = activity_template;
+
+    return (
+        <div
+            className="bcs-app-activity-container"
+            data-resin-target="loaded"
+            data-resin-feature={`app_activity_${templateId}_card`}
+        >
+            <div
+                className={classNames('bcs-app-activity', {
+                    'bcs-is-pending': isPending || error,
+                })}
+            >
+                <img className="bcs-app-activity-icon" alt={`${name} Icon`} src={icon_url} />
+                <span className="bcs-app-activity-content">
+                    <div className="bcs-app-activity-headline">
+                        <span className="bcs-app-activity-app-name">{name}</span>
+                        {(currentUser.id === created_by.id || canDelete) && !isPending && (
```

Comment:
Should the user id check be folded into the `canDelete` variable?

Reply:
Sure!